### PR TITLE
Convert most Promise chains to async/await

### PR DESCRIPTION
**Pull request by tylerturdenpants**

### CODE REVIEW

File: app/controllers/profile.js [hunk 1]
```diff
@@ -10,16 +10,20 @@ export default Controller.extend({
   followUser: task(function*(profile) {
     this.toggleProperty('waitingForFollowing');
 
-    yield profile.follow().finally(() => {
+    try {
+      yield profile.follow();
+    } finally {
       this.toggleProperty('waitingForFollowing');
-    });
+    }
```
Comment (Alonski) [inline app/controllers/profile.js]: Hmm looking at this again why are we even doing this?
Why not just have a computed property:
```js
waitingForFollowing: computed("followUser.isRunning", "unFollowUser.isRunning", {
  get() {
    return this.get("followUser.isRunning") || this.get("unFollowUser.isRunning");
  }
})
```

Does this make more sense?

**Comment by Alonski** [comment]
@tylerturdenpants Did you also yarn deduplicate?

File: app/controllers/profile.js (hunk 1 above)
Comment (tylerturdenpants) [reply app/controllers/profile.js]: Yes, but I was just apply best patterns not really changing the logic. I can fix if you want.

**Comment by tylerturdenpants** [comment]
Nope. I just ran yarn. This may be a byproduct of my yarn version as compared to the the version of the one that generated this file. I’ll run it again on my 1.19.2 version.

File: app/controllers/profile.js (hunk 1 above)
Comment (GCheung55) [reply app/controllers/profile.js]: Perhaps it would make sense to use the `or` computer macro instead of `computed`?

**Pull request merged**